subxt/src/client/online_client.rs

```diff
@@ -209,34 +233,107 @@ impl<T: Config> ClientRuntimeUpdater<T> {
         &curr.runtime_version != new
     }
 
+    fn do_update(&self, update: Update) {
+        let mut writable = self.0.inner.write();
+        writable.metadata = update.metadata;
+        writable.runtime_version = update.runtime_version;
+    }
+
+    /// Tries to apply a new update.
+    pub fn apply_update(&self, update: Update) -> Result<(), UpgradeError> {
+        if !self.is_runtime_version_different(&update.runtime_version) {
+            return Err(UpgradeError::SameVersion)
+        }
+
+        self.do_update(update);
+
+        Ok(())
+    }
+
     /// Performs runtime updates indefinitely unless encountering an error.
     ///
     /// *Note:* This will run indefinitely until it errors, so the typical usage
     /// would be to run it in a separate background task.
     pub async fn perform_runtime_updates(&self) -> Result<(), Error> {
         // Obtain an update subscription to further detect changes in the runtime version of the node.
-        let mut update_subscription = self.0.rpc.subscribe_runtime_version().await?;
-
-        while let Some(new_runtime_version) = update_subscription.next().await {
-            // The Runtime Version obtained via subscription.
-            let new_runtime_version = new_runtime_version?;
+        let mut runtime_version_stream = self.runtime_updates().await?;
 
-            // Ignore this update if there is no difference.
-            if !self.is_runtime_version_different(&new_runtime_version) {
-                continue
-            }
+        while let Some(update) = runtime_version_stream.next().await {
+            let update = update?;
 
-            // Fetch new metadata.
-            let new_metadata = self.0.rpc.metadata().await?;
-
-            // Do the update.
-            let mut writable = self.0.inner.write();
-            writable.metadata = new_metadata;
-            writable.runtime_version = new_runtime_version;
+            // Ignore if the update fails.
```

Comment:
```suggestion
            // This only fails if received the runtime version is the same the current runtime version
            // which might occur because that runtime subscriptions in substrate sends out the initial
            // value when they created and not only when runtime upgrades occurs.
            // Thus, fine to ignore here as it strictly speaking isn't really an error
```